newa/__init__.py

```diff
@@ -634,8 +663,9 @@ def fetch_details(self) -> None:
 class Execution(Cloneable, Serializable):
     """ A test job execution """
 
-    return_code: Optional[int] = None
+    return_code: Optional[int] = 0
```

Comment:
Does this field need to be `Optional` when the default is no longer `None`? It does not seem to be used anywhere but here and at https://github.com/RedHatQE/newa/pull/27/files#diff-fbfe615aa583117b7a57302f5da6ff45c1eda26a6f4b2406b68a09ffe1bd698aR343, where the value is even explicitly given. Does the field need to be `Optional` or even have a default value? It seems to be set to `0` right from the beginning when instance of `Execution` is created :)

The same seems to apply to `batch_id` - it has default value, but that value is never used. When `Execution` is instantiated, `batch_id` is set at that moment, https://github.com/RedHatQE/newa/pull/27/files#diff-fbfe615aa583117b7a57302f5da6ff45c1eda26a6f4b2406b68a09ffe1bd698aR345.

Both fields seem to be initialized by whoever creates an instance, therefore it would be safer to drop the default values. Then we can catch possible call sites where `Execution` is created without providing these values.

Reply:
Fixed.